app/src/main/java/com/picone/go4lunch/presentation/ui/main/MainActivity.java

```diff
@@ -90,7 +107,14 @@ private void initMenuButton() {
             mBinding.drawerLayout.close();
             switch (item.getItemId()) {
                 case R.id.your_lunch_drawer_layout:
-                    mNavController.navigate(R.id.restaurantDetailFragment);
+                    mRestaurantViewModel.getCurrentUser.observe(this,user -> {
+                        if (user.getUserDailySchedule()!=null){
```

Comment:
Same here. Nav should be launched when selectedRestaurant is set

Reply:
ok every navigation on live data subscription